### PR TITLE
Added typescript definitions for SideToolbar plugin

### PR DESCRIPTION
**Pull request by CFerndp**
Please check out Contributing Guidelines. By following this template you help us to review your code.
https://github.com/draft-js-plugins/draft-js-plugins/blob/master/CONTRIBUTING.md

## Checklist
- [X] Fix any eslint errors that occur
- [X] Update change-log for every plugin you touch
- [X] Add/Update tests if you add/change new functionality
- [X] Add/Update docs if you add/change functionality
- [X] Enable "Allow edits from maintainers" for this PR

## Use-case/Problem
Not able to use this plugin within a TypeScript project

## Implementation
Added `index.d.ts` file with a definition of each part of the plugin

### CODE REVIEW

File: draft-js-side-toolbar-plugin/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 - Hide internals in single bundle
 - Add esm support
 
+## 3.0.3
+- Added TS definitions
```
Comment (TrySound) [inline draft-js-side-toolbar-plugin/CHANGELOG.md]: Put it in to be released section please

File: draft-js-side-toolbar-plugin/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "draft-js-side-toolbar-plugin",
-  "version": "3.0.2",
+  "version": "3.0.3",
```
Comment (TrySound) [inline draft-js-side-toolbar-plugin/package.json]: Do not increase version in this PR. We will do that separately.

Comment (CFerndp) [reply draft-js-side-toolbar-plugin/package.json]: Sure, sorry about it

File: draft-js-side-toolbar-plugin/src/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+export interface SideToolbarChildrenProps {
+    theme: SideToolbarPluginTheme["buttonStyles"];
+    getEditorState: () => EditorState;
+    setEditorState: (editorState: EditorState) => void;
+    onOverrideContent: (content: ComponentType<SideToolbarChildrenProps>) => void;
+}
+
+export interface SideToolbarProps {
+    children?: ComponentType<SideToolbarChildrenProps>;
```
Comment (TrySound) [inline draft-js-side-toolbar-plugin/src/index.d.ts]: This should be a function, not a component.

Comment (CFerndp) [reply draft-js-side-toolbar-plugin/src/index.d.ts]: it's the same, a function who receive props and return a ReactNode is the same that if I defined it as a ComponentType who receive props typed by SideToolbarChildrenProps

Comment (TrySound) [reply draft-js-side-toolbar-plugin/src/index.d.ts]: No, it's not.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L71

This function is just called. Not instantiated with React.createElement. If you use ComponentType user can pass class components. Classes cannot be called without `new` keyword.
https://github.com/draft-js-plugins/draft-js-plugins/blob/040436205cdf59236267c4dca40c80d41b8b8f9c/draft-js-side-toolbar-plugin/src/components/BlockTypeSelect/index.js#L62

Comment (CFerndp) [reply draft-js-side-toolbar-plugin/src/index.d.ts]: Totally right, fixed. Thank you so much for the information and the effort :)

**Pull request merged**